src/framework/coercion/core.clj

```diff
@@ -48,10 +50,18 @@
    :error (fn [state]
             (xiana/error (assoc state :response {:status 400
                                                  :body   "Request coercion failed"})))
-   :leave (fn [{{:keys [:status :body]} :response
-                :as                     state}]
-            (let [schema (get-in state [:request-data :match :data :responses status :body])]
-              (cond (and schema body (m/validate schema body)) (xiana/ok state)
-                    (and schema body) (xiana/error (assoc state :response {:status 400
-                                                                           :body   "Response validation failed"}))
-                    :else (xiana/ok state))))})
+   :leave (fn [{{:keys [:status :body]}  :response
+                {method :request-method} :request
+                :as                      state}]
+            (let [schema (or (get-in state [:request-data :match :data method :responses status :body])
+                             (get-in state [:request-data :match :data :responses status :body]))
+                  valid? (some-> schema (m/validate body))]
+              (if (false? valid?)
+                (let [explain (m/explain schema body)
+                      humanized (me/humanize explain)]
+                  (log/warn "Response validation FAILED")
```

Comment:
this can also be a request validation failure, right?

Reply:
I don't think it's a good idea to log failed requests. In most cases an error response with useful message will be more than enough.